### PR TITLE
Wappalyzer - make data available for reports

### PR DESCRIPTION
**Pull request by psiinon**
I've not added any help as I'm not sure where it would or should go :/
I've also not added a section for it - if the datas available then it will be included - the add-on is optional after all.

Example report snippet:
Screenshot_2021-05-18 ZAP Scanning Report

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Include the icons? (I have no idea how much effort that would be.)

**Comment by psiinon** [comment]
@kingthorin not that easy tbh - resources like images are stored with report templates, there is no mechanism for them to be read from data. In theory an HTML report could have no resources in which case it wouldnt create a subdirectory for us to put any resources in.

**Comment by kingthorin** [comment]
Could we get the ImageIcon from Wappalyzer and turn it into an inline image with a data URL?

**Comment by psiinon** [comment]
In theory yes :)

**Comment by kingthorin** [comment]
Okay, not a blocker :)

**Comment by psiinon** [comment]
Done.

**Review by kingthorin — APPROVED** [review]

File: addOns/automation/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Added
 - Support for using variables in the config.
+- Support for null jobs.
```
Comment (thc202) [inline addOns/automation/CHANGELOG.md:9]: null → data?

File: addOns/reports/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ All notable changes to this add-on will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ## Unreleased
+### Added
+- Wappalyser data to the traditional-html-plus report, if its available.
```
Comment (thc202) [inline addOns/reports/CHANGELOG.md:8]: - `Wappalyzer`
 - `it's`

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/automation/ExtensionWappalyzerAutomation.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    public static final String NAME = "ExtensionWappalyzerAutomation";
+
+    private WappalyzerJob job;
+
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(2);
+        dependencies.add(ExtensionAutomation.class);
```
Comment (thc202) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/automation/ExtensionWappalyzerAutomation.java:42]: Should also depend on `ExtensionWappalyzer`.

**Comment by thc202** [comment]
The help could be added top the Wappalyzer add-on(?), a paragraph saying that the information is available to reports would be enough IMO.

**Comment by psiinon** [comment]
Hopefully fixed those comments :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by psiinon** [comment]
Merging as LGTM appears to be stuck. It didnt report any error in the last run and the only changes after that were docs or v minor...

**Pull request merged**